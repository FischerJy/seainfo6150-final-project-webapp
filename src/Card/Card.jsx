import React from "react";
import { Link } from "react-router-dom";

export default function Card(info) {
  return (
    <div className="card">
      <p>name:{info.name}</p>
      <p>lat:{info.coord.lat}</p>
      <p>lon:{info.coord.lon}</p>
      <br />
      <div>
        wind
        <p>speed:{info.wind.speed}</p>
        <p>deg:{info.wind.deg}</p>
      </div>
      <br />
      <div>
        temp
        <p>temp:{info.main.temp}</p>
        <p>temp_max:{info.main.temp_max}</p>
        <p>temp_min:{info.main.temp_min}</p>
      </div>
      <br />
      <div>
        weather
        <p>main:{info.weather[0].main}</p>
        <p>description:{info.weather[0].description}</p>
        <p>
          <Link
            to={`/hours/${info.name}/${info.coord.lat}/${info.coord.lon}/${info.dt}`}
          >
            Hours Detail
          </Link>
        </p>
      </div>
    </div>
  );
}
