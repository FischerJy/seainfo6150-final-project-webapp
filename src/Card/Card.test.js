import React from "react";
import { render } from "@testing-library/react";
import Card from "./Card.jsx";
import { BrowserRouter } from "react-router-dom";

var obj = {
    wind: {},
    coord: { lat: '', lon: '' },
    main: {},
    weather: [{}],
}

it("renders the Card component correctly", () => {
    const { container } = render(<BrowserRouter>
        <Card {...obj} />
    </BrowserRouter>);
    expect(container).toMatchSnapshot();
});