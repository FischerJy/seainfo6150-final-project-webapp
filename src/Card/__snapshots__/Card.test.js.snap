// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Card component correctly 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      name:
    </p>
    <p>
      lat:
      
    </p>
    <p>
      lon:
      
    </p>
    <br />
    <div>
      wind
      <p>
        speed:
      </p>
      <p>
        deg:
      </p>
    </div>
    <br />
    <div>
      temp
      <p>
        temp:
      </p>
      <p>
        temp_max:
      </p>
      <p>
        temp_min:
      </p>
    </div>
    <br />
    <div>
      weather
      <p>
        main:
      </p>
      <p>
        description:
      </p>
      <p>
        <a
          href="/hours/undefined///undefined"
        >
          Hours Detail
        </a>
      </p>
    </div>
  </div>
</div>
`;
